apps/meteor/app/api/server/v1/im.ts

```diff
@@ -123,18 +123,35 @@ API.v1.addRoute(
 			if (!roomId) {
 				throw new Meteor.Error('error-room-param-not-provided', 'Body param "roomId" is required');
 			}
-			const canAccess = await canAccessRoomIdAsync(roomId, this.userId);
-			if (!canAccess) {
-				return API.v1.unauthorized();
+
+			let rid: IRoom['_id'] = roomId;
+			let subscription: ISubscription | null = null;
+
+			const roomExists = !!(await Rooms.findOneById(roomId));
+			if (!roomExists) {
+				// if the room doesnt exists, should check if the user has a subscription to it, and if has, close it
+				subscription = await Subscriptions.findOneByRoomIdAndUserId(roomId, this.userId);
+			} else {
+				const canAccess = await canAccessRoomIdAsync(roomId, this.userId);
+				if (!canAccess) {
+					return API.v1.unauthorized();
+				}
+
+				const { room, subscription: subs } = await findDirectMessageRoom({ roomId }, this.userId);
+
+				rid = room._id;
+				subscription = subs;
 			}
 
-			const { room, subscription } = await findDirectMessageRoom({ roomId }, this.userId);
+			if (!subscription) {
+				return API.v1.failure(`The user is not subscribed to the room`);
+			}
 
-			if (!subscription?.open) {
+			if (!subscription.open) {
 				return API.v1.failure(`The direct message room, is already closed to the sender`);
 			}
 
-			await hideRoomMethod(this.userId, room._id);
+			await hideRoomMethod(this.userId, rid);
```

Comment:
```suggestion
			await hideRoomMethod(this.userId, roomId);
```
Saving one variable here, let's use the one we already have.